bin/sce_qc_report.R

```diff
@@ -233,14 +248,45 @@ if (multiplexed) {
 }
 
 # Output metadata as JSON
-jsonlite::write_json(metadata_list, path = opt$metadata_json, auto_unbox = TRUE, pretty = TRUE)
+jsonlite::write_json(
+  metadata_list, 
+  path = opt$metadata_json, 
+  auto_unbox = TRUE, 
+  pretty = TRUE
+)
 
+# render main QC report
 scpcaTools::generate_qc_report(
   library_id = metadata_list$library_id,
   unfiltered_sce = unfiltered_sce,
   filtered_sce = filtered_sce,
   processed_sce = processed_sce,
   report_template = opt$report_template,
   output = opt$qc_report_file,
-  extra_params = list(seed = opt$seed)
+  extra_params = list(
+    seed = opt$seed,
+    # this will only be used if cell types exist
+    celltype_report = opt$celltype_report_file
+  )
 )
+
+
+# render supplemental cell types report, if needed
+if (opt$celltype_report_file != "") {
+  
+  # check that the template file exists
+  if (!file.exists(opt$celltype_report_template)) {
+    stop("Supplemental cell types report template not found.")
+  }
+  
+  # render report
+  rmarkdown::render(
+    input = opt$celltype_report_template,
+    output_file = opt$celltype_report_file,
```

Comment:
A few safety things copied over from `generate_qc_report`.
My recollection is that there are some weird decisions about where output files are rendered (in the template directory?) so we have to be explicit about both the output file and input files.

And while we are at it, we might as well use tempdirs for everything beforehand and a fresh environment.
```suggestion
    output_file = basename(opt$celltype_report_file),
    output_dir = dirname(opt$celltype_report_file),
    intermediates_dir = tempdir(),
    knit_root_dir = tempdir(),
    envir = new.env(),
```